lib/mixins/vector.js

```diff
@@ -63,14 +63,18 @@ export default {
 
   dash(length, options = {}) {
     let phase;
-    if (length == null) {
-      return this;
-    }
     if (Array.isArray(length)) {
+      const hasZeros = length.filter(x => x === 0).length > 0;
+      if(hasZeros) {
+        throw new Error(`dash(${JSON.stringify(length)}) invalid, lengths must be greater than zero`);
+      }
       length = length.map(v => number(v)).join(' ');
       phase = options.phase || 0;
       return this.addContent(`[${length}] ${number(phase)} d`);
     } else {
+      if (!Array.isArray(length) && !Number.isInteger(length) || length === 0) {
+        throw new Error(`dash(${length}) invalid, must be number greater than zero`);
```

Comment:
1. `!Array.isArray(length)` has already been checked and will always return true. I think it should be omitted here.
2. The length doesn't have to be an integer. The PDF spec allows floating point numbers. `Number.isFinite(length)` should get you your null check while also covering cases like `NaN` and strings and infinities.
3. Your check is only for zero. You should also check for negative numbers.

Reply:
Agreed, nice catches. I went ahead and consolidated the parallel paths in the `if` and `else` - seems cleaner now.